src/Orleans/Statistics/CounterStatistic.cs

```diff
@@ -179,11 +185,27 @@ public void IncrementBy(long n)
         {
             if (isOrleansManagedThread)
             {
-                while (perOrleansThreadCounters.Count <= id)
+                ReferenceLong orleansValue = null;
+               
+                try
```

Comment:
I wanted to save the check on every increment of a counter, exception will happen just couple times per thread but check will happen constantly.